sdk/src/main/java/io/customer/sdk/queue/QueueStorage.kt

```diff
@@ -0,0 +1,102 @@
+package io.customer.sdk.queue
+
+import io.customer.sdk.data.store.FileStorage
+import io.customer.sdk.data.store.FileType
+import io.customer.sdk.queue.type.QueueInventory
+import io.customer.sdk.queue.type.QueueModifyResult
+import io.customer.sdk.queue.type.QueueStatus
+import io.customer.sdk.queue.type.QueueTask
+import io.customer.sdk.queue.type.QueueTaskMetadata
+import io.customer.sdk.queue.type.QueueTaskRunResults
+import io.customer.sdk.util.JsonAdapter
+import java.util.*
+
+interface QueueStorage {
+    fun getInventory(): QueueInventory
+    fun saveInventory(inventory: QueueInventory): Boolean
+    fun create(type: String, data: String): QueueModifyResult
+    fun get(taskStorageId: String): QueueTask?
+    fun delete(taskStorageId: String): QueueModifyResult
```

Comment:
What happens if we switch `siteID` are we going to loop and delete all tasks?

Reply:
At this time, no. Once a task gets added to the background queue, it will get saved to the file system in it's own "siteid sandbox" (a directory of files in the file system only belonging to that siteid). We only delete tasks from the background queue if a task runs successfully. 

We have [had discussions](https://github.com/customerio/issues/issues/6793) about doing something like this in the future but we have not made any decisions on moving forward with doing that.